pkg/odo/cli/component/list.go

```diff
@@ -47,27 +56,51 @@ func NewListOptions() *ListOptions {
 // Complete completes log args
 func (lo *ListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 
-	if util.CheckKubeConfigExist() {
-		klog.V(4).Infof("New Context")
-		lo.Context = genericclioptions.NewContext(cmd)
+	if experimental.IsExperimentalModeEnabled() {
+		// Add a disclaimer that we are in *experimental mode*
+		log.Experimental("Experimental mode is enabled, use at your own risk")
+
+		lo.Context = genericclioptions.NewDevfileContext(cmd)
+
 	} else {
-		klog.V(4).Infof("New Config Context")
-		lo.Context = genericclioptions.NewConfigContext(cmd)
+		if util.CheckKubeConfigExist() {
+			klog.V(4).Infof("New Context")
+			lo.Context = genericclioptions.NewContext(cmd)
+		} else {
+			klog.V(4).Infof("New Config Context")
+			lo.Context = genericclioptions.NewConfigContext(cmd)
 
+		}
 	}
+
+	lo.Client = genericclioptions.Client(cmd)
+	dcSupport, err := lo.Client.IsDeploymentConfigSupported()
+
+	if err != nil {
+		return err
+	}
+
+	lo.hasDCSupport = dcSupport
+
 	return
 
 }
 
 // Validate validates the list parameters
 func (lo *ListOptions) Validate() (err error) {
 
-	var project, app string
-
 	if len(lo.Application) != 0 && lo.allAppsFlag {
 		klog.V(4).Infof("either --app and --all-apps both provided or provided --all-apps in a folder has app, use --all-apps anyway")
 	}
 
+	if experimental.IsExperimentalModeEnabled() {
```

Comment:
maybe store the value somewhere and use it here

Reply:
done